From 0793db96f20862a3993ccaa1e670b6d44f5e3d17 Mon Sep 17 00:00:00 2001
From: Andra-Teodora Ilie <andra.ilie@nxp.com>
Date: Fri, 26 Nov 2021 14:42:13 +0200
Subject: [PATCH 13/50] s32g: clocks: Replace shared serdes1_lane1 clocks with
 serdes1_lane0

This is a prerequisite for adding s32r45 clocks. Clocks were replaced
in U-Boot as part of commit b8055e24bd10172a5416f4f811d707de98fdaf96.

Issue: ALB-7411
Upstream-Status: Pending 

Signed-off-by: Andra-Teodora Ilie <andra.ilie@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/nxp/s32/clk/s32g_clk.c                | 20 +++++++++----------
 drivers/nxp/s32/clk/s32gen1_clk.c             | 19 +++++++++---------
 .../drivers/nxp/s32/clk/s32gen1_shared_clks.h |  4 ++--
 3 files changed, 21 insertions(+), 22 deletions(-)

diff --git a/drivers/nxp/s32/clk/s32g_clk.c b/drivers/nxp/s32/clk/s32g_clk.c
index 97d9b4261..7ccd01baa 100644
--- a/drivers/nxp/s32/clk/s32g_clk.c
+++ b/drivers/nxp/s32/clk/s32g_clk.c
@@ -33,11 +33,6 @@ static struct s32gen1_fixed_clock pfe_mac0_ext_ref =
 static struct s32gen1_clk pfe_mac0_ext_ref_clk =
 		S32GEN1_MODULE_CLK(pfe_mac0_ext_ref);
 
-static struct s32gen1_fixed_clock serdes1_lane0_ext_tx =
-		S32GEN1_FIXED_CLK_INIT();
-static struct s32gen1_clk serdes1_lane0_ext_tx_clk =
-		S32GEN1_MODULE_CLK(serdes1_lane0_ext_tx);
-
 static struct s32gen1_mux cgm2_mux1 =
 		S32GEN1_MUX_INIT(S32GEN1_CGM2, 1, 5,
 				 S32GEN1_CLK_FIRC,
@@ -71,11 +66,6 @@ static struct s32gen1_clk pfe_mac0_ref_div_clk =
 		S32GEN1_FREQ_MODULE_CLK(pfe0_ref_block, 0, 50 * MHZ);
 
 /* PFE_MAC0_RX */
-static struct s32gen1_fixed_clock serdes1_lane0_cdr =
-		S32GEN1_FIXED_CLK_INIT();
-static struct s32gen1_clk serdes1_lane0_cdr_clk =
-		S32GEN1_MODULE_CLK(serdes1_lane0_cdr);
-
 static struct s32gen1_fixed_clock pfe_mac0_ext_rx =
 		S32GEN1_FIXED_CLK_INIT();
 static struct s32gen1_clk pfe_mac0_ext_rx_clk =
@@ -96,6 +86,10 @@ static struct s32gen1_clk pfe_mac0_rx_clk =
 		S32GEN1_FREQ_MODULE_CLK(pfe0_rx_block, 2500000, 312500000);
 
 /* PFE_MAC1_TX */
+static struct s32gen1_fixed_clock serdes1_lane1_tx =
+		S32GEN1_FIXED_CLK_INIT();
+struct s32gen1_clk serdes1_lane1_tx_clk =
+		S32GEN1_MODULE_CLK(serdes1_lane1_tx);
 static struct s32gen1_fixed_clock pfe_mac1_ext_tx =
 		S32GEN1_FIXED_CLK_INIT();
 static struct s32gen1_clk pfe_mac1_ext_tx_clk =
@@ -139,6 +133,10 @@ static struct s32gen1_clk pfe_mac1_ref_div_clk =
 		S32GEN1_FREQ_MODULE_CLK(pfe1_ref_block, 0, 50 * MHZ);
 
 /* PFE_MAC1_RX */
+static struct s32gen1_fixed_clock serdes1_lane1_cdr =
+		S32GEN1_FIXED_CLK_INIT();
+struct s32gen1_clk serdes1_lane1_cdr_clk =
+		S32GEN1_MODULE_CLK(serdes1_lane1_cdr);
 static struct s32gen1_fixed_clock pfe_mac1_ext_rx =
 		S32GEN1_FIXED_CLK_INIT();
 static struct s32gen1_clk pfe_mac1_ext_rx_clk =
@@ -290,7 +288,7 @@ static struct s32gen1_clk *s32g_cc_clocks[] = {
 	[CC_ARR_CLK(S32G_CLK_PFE_MAC2_EXT_TX)] = &pfe_mac2_ext_tx_clk,
 	[CC_ARR_CLK(S32G_CLK_PFE_MAC2_EXT_RX)] = &pfe_mac2_ext_rx_clk,
 	[CC_ARR_CLK(S32G_CLK_PFE_MAC2_EXT_REF)] = &pfe_mac2_ext_ref_clk,
-	[CC_ARR_CLK(S32G_CLK_SERDES1_LANE0_TX)] = &serdes1_lane0_ext_tx_clk,
+	[CC_ARR_CLK(S32G_CLK_SERDES1_LANE0_TX)] = &serdes1_lane0_tx_clk,
 	[CC_ARR_CLK(S32G_CLK_SERDES1_LANE0_CDR)] = &serdes1_lane0_cdr_clk,
 	[CC_ARR_CLK(S32G_CLK_PFE_MAC0_REF_DIV)] = &pfe_mac0_ref_div_clk,
 	[CC_ARR_CLK(S32G_CLK_PFE_MAC1_REF_DIV)] = &pfe_mac1_ref_div_clk,
diff --git a/drivers/nxp/s32/clk/s32gen1_clk.c b/drivers/nxp/s32/clk/s32gen1_clk.c
index 102123b42..f52b33ad2 100644
--- a/drivers/nxp/s32/clk/s32gen1_clk.c
+++ b/drivers/nxp/s32/clk/s32gen1_clk.c
@@ -16,15 +16,6 @@ static struct s32gen1_osc firc =
 static struct s32gen1_osc sirc =
 		S32GEN1_OSC_INIT(S32GEN1_SIRC);
 
-static struct s32gen1_fixed_clock serdes1_lane1_tx =
-		S32GEN1_FIXED_CLK_INIT();
-struct s32gen1_clk serdes1_lane1_tx_clk =
-		S32GEN1_MODULE_CLK(serdes1_lane1_tx);
-static struct s32gen1_fixed_clock serdes1_lane1_cdr =
-		S32GEN1_FIXED_CLK_INIT();
-struct s32gen1_clk serdes1_lane1_cdr_clk =
-		S32GEN1_MODULE_CLK(serdes1_lane1_cdr);
-
 /* Muxes */
 static struct s32gen1_mux arm_pll_mux =
 	S32GEN1_MUX_INIT(S32GEN1_ARM_PLL, 0, 2,
@@ -331,6 +322,16 @@ static struct s32gen1_fixed_clock serdes0_lane0_cdr =
 static struct s32gen1_clk serdes0_lane0_cdr_clk =
 		S32GEN1_MODULE_CLK(serdes0_lane0_cdr);
 
+/* S32GEN1_CLK_SERDES1_LANE0_TX */
+static struct s32gen1_fixed_clock serdes1_lane0_tx =
+		S32GEN1_FIXED_CLK_INIT();
+struct s32gen1_clk serdes1_lane0_tx_clk =
+		S32GEN1_MODULE_CLK(serdes1_lane0_tx);
+static struct s32gen1_fixed_clock serdes1_lane0_cdr =
+		S32GEN1_FIXED_CLK_INIT();
+struct s32gen1_clk serdes1_lane0_cdr_clk =
+		S32GEN1_MODULE_CLK(serdes1_lane0_cdr);
+
 /* SPI_CLK */
 static struct s32gen1_mux cgm0_mux16 =
 		S32GEN1_MUX_INIT(S32GEN1_CGM0, 16, 2,
diff --git a/include/drivers/nxp/s32/clk/s32gen1_shared_clks.h b/include/drivers/nxp/s32/clk/s32gen1_shared_clks.h
index 6cda1b40d..222916caa 100644
--- a/include/drivers/nxp/s32/clk/s32gen1_shared_clks.h
+++ b/include/drivers/nxp/s32/clk/s32gen1_shared_clks.h
@@ -10,8 +10,8 @@ extern struct s32gen1_clk xbar_2x_clk;
 extern struct s32gen1_cgm_div per_div;
 extern struct s32gen1_clk cgm0_mux0_clk;
 extern struct s32gen1_dfs armdfs;
-extern struct s32gen1_clk serdes1_lane1_tx_clk;
-extern struct s32gen1_clk serdes1_lane1_cdr_clk;
+extern struct s32gen1_clk serdes1_lane0_tx_clk;
+extern struct s32gen1_clk serdes1_lane0_cdr_clk;
 extern struct s32gen1_clk accel_pll_phi0_clk;
 extern struct s32gen1_clk accel_pll_phi1_clk;
 extern const struct siul2_freq_mapping siul2_clk_freq_map[];
-- 
2.17.1

